tenable/ad/profiles/api.py

```diff
@@ -0,0 +1,196 @@
+'''
+Profiles
+=============
+
+Methods described in this section relate to the profiles API.
+These methods can be accessed at ``TenableAD.profiles``.
+
+.. rst-class:: hide-signature
+.. autoclass:: ProfilesAPI
+    :members:
+'''
+from typing import List, Dict
+from tenable.base.endpoint import APIEndpoint
+from .schema import ProfileSchema
+
+
+class ProfilesAPI(APIEndpoint):
+    _path = 'profiles'
+    _schema = ProfileSchema()
+
+    def list(self) -> List[Dict]:
+        '''
+        Retrieve all profiles
+
+        Returns:
+            list[dict]:
+                The list of profile objects
+
+        Examples:
+            >>> tad.profiles.list()
+        '''
+        return self._schema.load(self._get(), many=True)
+
+    def create(self,
+               name: str,
+               directories: List[int]
+               ) -> List[Dict]:
+        '''
+        Create a profile
+
+        Args:
+            name (str):
+                The name of new profile.
+            directories (List[int]):
+                The list of directory identifiers.
+
+        Return:
+            list[dict]:
+                The created profile objects
+
+        Example:
+            >>> tad.profiles.create(
+            ...     name='ExampleProfile',
+            ...     directories=[1, 2]
+            ...     )
+        '''
+        payload = [self._schema.dump({
+            'name': name,
+            'directories': directories
+        })]
+
+        return self._schema.load(self._post(json=payload), many=True)
+
+    def details(self,
+                profile_id: str
+                ) -> Dict:
+        '''
+        Retrieves the details for a specific profile
+
+        Args:
+            profile_id (str):
+                The profile instance identifier.
+
+        Returns:
+            dict:
+                The profile object.
+
+        Examples:
+            >>> tad.profiles.details('1')
+        '''
+        return self._schema.load(self._get(f'{profile_id}'))
+
+    def update(self,
+               profile_id: str,
+               **kwargs
+               ) -> Dict:
+        '''
+        Update an existing profile
+
+        Args:
+            profile_id (str):
+                The profile instance identifier.
+            name (optional, str):
+                The name of profile.
+            deleted (optional, bool):
+                is the profile deleted?
+            directories (optional, List[int]):
+                The list of directory ids.
+
+        Returns:
+            dict:
+                The updated profile object.
+
+        Examples:
+            >>> tad.profiles.update(
+            ...     profile_id='1',
+            ...     name='EDITED'
+            ...     )
+        '''
+        payload = self._schema.dump(kwargs)
```

Comment:
should load the schema before dumping to perform validation.

Reply:
update request payload with `schema.load` before dumping.